middleware/context_timeout.go

```diff
@@ -0,0 +1,89 @@
+package middleware
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/labstack/echo/v4"
+)
+
+// ContextTimeoutConfig defines the config for ContextTimeout middleware.
+type ContextTimeoutConfig struct {
+	// Skipper defines a function to skip middleware.
+	Skipper Skipper
+
+	// ErrorHandler is a function when error aries in middeware execution.
+	ErrorHandler func(err error, c echo.Context) error
+
+	// Timeout configures a timeout for the middleware, defaults to 0 for no timeout
+	Timeout time.Duration
+}
+
+var (
+	// DefaultContextTimeoutErrorHandler is default error handler of ContextTimeout middleware.
+	DefaultContextTimeoutErrorHandler = func(err error, c echo.Context) error {
+		if err != nil {
+			if errors.Is(err, context.DeadlineExceeded) {
+				return echo.ErrServiceUnavailable
+			}
+			return err
+		}
+		return nil
+	}
+)
+
+// ContextTimeout returns a middleware which returns error (503 Service Unavailable error) to client
+// when underlying method returns context.DeadlineExceeded error.
+func ContextTimeout(timeout time.Duration) echo.MiddlewareFunc {
+	config := ContextTimeoutConfig{
+		Skipper:      DefaultSkipper,
```

Comment:
`Skipper` and `ErrorHandler` does not need to be set here. `ToMiddleware` handles these.

This would be probably fine:
```go
func ContextTimeout(timeout time.Duration) echo.MiddlewareFunc {
	return ContextTimeoutWithConfig(ContextTimeoutConfig{Timeout: timeout})
}
```